design-proposals/klip-43-timestamp-data-type-support.md

```diff
@@ -0,0 +1,159 @@
+# KLIP-43: TIMESTAMP Data Type Support
+
+**Author**: @jzaralim | 
+**Release Target**: 0.15 | 
+**Status**: In Discussion | 
+**Discussion**: https://github.com/confluentinc/ksql/pull/6649
+
+**tl;dr:** _KSQL currently converts time data into long values. This KLIP introduces a TIMESTAMP
+data type to handle time data instead._
+
+## Motivation and background
+
+With Connect integration, KSQL processes data from various sources and sinks. Most other databases
+have a TIMESTAMP or DATETIME type, which KSQL currently converts into long values. This sometimes
+causes problems when sinking into a database. Supporting TIMESTAMP types would make moving time data
+between KSQL and other data sources/sinks smoother and less error-prone.
+
+## What is in scope
+### Required
+
+* Add TIMESTAMP type to KSQL
+* Serialization and de-serialization of TIMESTAMPs to Avro, JSON and Delimited formats
+* Built-in functions to support TIMESTAMP usage
+
+### Future enhancements
+
+* Give window units (HOUR, DAY etc) a TIMESTAMP value
+* Add DATE and TIME types
+* UDFs such as DAY and HOUR
+
+## What is not in scope
+Changing the ROWTIME data type. We will eventually want this to happen, but that is a separate discussion.
+
+## Public APIS
+
+The TIMESTAMP data type will store a time without timezone information. It will be useable in any
+place STRING or INTEGER could be used:
+```roomsql
+CREATE STREAM stream_name (col1 TIMESTAMP, COL2 STRING) AS ...
+CREATE TABLE table_name (col1 STRUCT<field TIMESTAMP>) AS ...
+```
+
+## Design
```

Comment:
I know this is trivial, but could you add the syntax how this would work? It's part of the design.

Also, what is a timestamp value? Is it a long value? So far, I read that ksqldB treats timestamps as long values, and that we don't what that (even if we store long values), so is there a different value to refer to timestamps?

Some SQL use this:
```
create or replace table time (ltz timestamp);
insert into time values ('2016-05-01 00:00:00.000');
```

We should clarify how a timestamp value will be defined.